lib/compilers/csharp.js

```diff
@@ -0,0 +1,151 @@
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        const programDir = path.dirname(inputFileName);
+        const clrBuildDir = '/home/ubuntu/runtime/artifacts/bin/coreclr/Linux.x64.Checked';
+
+        const projectFilePath = path.join(programDir, 'Program.csproj');
+        const coreClrJitPath = path.join(clrBuildDir, 'libclrjit.so');
+        const crossgenPath = path.join(clrBuildDir, 'crossgen');
+
+        const programPublishPath = path.join(
+            programDir,
+            'bin',
+            this.buildConfig,
+            this.targetFramework,
+            this.rID,
+            'publish',
+        );
+
+        const programDllPath = path.join(programPublishPath, 'Program.dll');
+
+        const projectFileContent =
+        `<Project Sdk="Microsoft.NET.Sdk">
+            <PropertyGroup>
+         <TargetFramework>${this.targetFramework}</TargetFramework>
+         <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+         </PropertyGroup>
+         </Project>
+        `;
+
+        execOptions.customCwd = programDir;
+        fs.writeFile(projectFilePath, projectFileContent);
+        options = this.compilerOptions;
+
+        const compilerResult = await super.runCompiler(
+            compiler,
+            options,
+            inputFileName,
+            execOptions,
+        );
+        const crossgenResult = await this.runCrossgen(
+            execOptions,
+            crossgenPath,
+            coreClrJitPath,
+            programPublishPath,
+            programDllPath,
+            this.getOutputFilename(programDir, ''),
+        );
+
+        if (crossgenResult.code !== 0) {
+            return crossgenResult;
+        }
+
+        return compilerResult;
+    }
+
+    optionsForFilter(_filters, _outputFilename, _userOptions) {
+        return this.compilerOptions;
+    }
+
+    getOutputFilename(dirPath, _outputFileBase) {
+        return path.join(dirPath, `output.s`);
+    }
+
+    cleanAsm(asm) {
+        var cleanedAsm = '';
+        var asmLines = asm.split('\n');
+
+        for (const line of asmLines) {
+            if (line.startsWith('; Assembly listing for method')) {
+                cleanedAsm = cleanedAsm.concat(line.substring(30) + ':\n');
+                continue;
+            }
+
+            if (line.startsWith('Microsoft (R)') ||
+                line.startsWith('Copyright (c)') ||
+                line.startsWith('Native image'))
+            {
+                continue;
+            }
+
+            cleanedAsm = cleanedAsm.concat(line.replace(/ +[\dA-Z]+ +[^\dA-Za-z]/,' ') + '\n');
+        }
+
+        return cleanedAsm;
+    }
+
+    async runCrossgen(execOptions, crossgenPath, jitPath, publishPath, dllPath, outputPath) {
+        execOptions.env['COMPlus_NgenDisasm'] = '*';
+        execOptions.env['COMPlus_NgenDiffableDasm'] = '1';
+
+        const crossgenOutput = await exec.execute(
```

Comment:
this should be `this.exec` instead of `exec.execute`. 

`exec` is defined on BaseCompiler to simply call `exec.execute`, but this may not always be the case in the future.